lib/services/assets-generation/assets-generation-service.ts

```diff
@@ -0,0 +1,109 @@
+import * as Jimp from "jimp";
+import * as Color from "color";
+import { exported } from "../../common/decorators";
+
+export const enum Operations {
+	OverlayWith = "overlayWith",
+	Blank = "blank",
+	Resize = "resize"
+}
+
+export class AssetsGenerationService implements IAssetsGenerationService {
+	private get propertiesToEnumerate(): IDictionary<string[]> {
+		return {
+			icon: ["icons"],
+			splash: ["splashBackgrounds", "splashCenterImages", "splashImages"]
+		};
+	}
+
+	constructor(private $logger: ILogger,
+		private $projectDataService: IProjectDataService) {
+	}
+
+	@exported("assetsGenerationService")
+	public async generateIcons(resourceGenerationData: IResourceGenerationData): Promise<void> {
+		this.$logger.info("Generating icons ...");
+		await this.generateImagesForDefinitions(resourceGenerationData, this.propertiesToEnumerate.icon);
+		this.$logger.info("Icons generation completed.");
+	}
+
+	@exported("assetsGenerationService")
+	public async generateSplashScreens(splashesGenerationData: ISplashesGenerationData): Promise<void> {
+		this.$logger.info("Generating splash screens ...");
+		await this.generateImagesForDefinitions(splashesGenerationData, this.propertiesToEnumerate.splash);
+		this.$logger.info("Splash screens generation completed.");
+	}
+
+	private async generateImagesForDefinitions(generationData: ISplashesGenerationData, propertiesToEnumerate: string[]): Promise<void> {
+		generationData.background = generationData.background || "white";
+		const assetsStructure = await this.$projectDataService.getAssetsStructure(generationData);
+
+		const assetItems = _(assetsStructure)
+			.filter((assetGroup: IAssetGroup, platform: string) => {
+				return !generationData.platform || platform.toLowerCase() === generationData.platform.toLowerCase();
+			})
+			.map((assetGroup: IAssetGroup) =>
+				_.filter(assetGroup, (assetSubGroup: IAssetSubGroup, imageTypeKey: string) =>
+					propertiesToEnumerate.indexOf(imageTypeKey) !== -1 && !assetSubGroup[imageTypeKey]
+				)
+			)
+			.flatten<IAssetSubGroup>()
+			.map(assetSubGroup => assetSubGroup.images)
+			.flatten<IAssetItem>()
+			.filter(assetItem => !!assetItem.filename)
+			.value();
+
+		for (const assetItem of assetItems) {
+			const operation = assetItem.resizeOperation || Operations.Resize;
```

Comment:
How do you infer the image operation for the asset only from ``Content.json``?

Reply:
It is taken in project-data-service for Android, but I've missed to set it for iOS :crying_cat_face: